lightpath/tests/test_widgets.py

```diff
@@ -27,15 +28,32 @@ def test_widget_updates(lightrow: LightRow, path: BeamPath):
     ipimb.insert()
     ipimb.remove()
     ipimb.insert()
-    assert (to_stylesheet_color(state_colors['half_removed'])
-            in ipimb_row.state_label.styleSheet())
+
+    def half_removed():
+        return (to_stylesheet_color(state_colors['half_removed'])
+                in ipimb_row.state_label.styleSheet())
+
+    wait_until(half_removed)
+    assert half_removed
```

Comment:
I think these are all typos, you meant to assert the function result, right?
```suggestion
    assert half_removed()
```

Reply:
hah no wonder it never failed 🤦

Reply:
Interestingly, locally the `removed` case can seemingly hang forever (10+ seconds, which is effectively forever for a computer brain)

Will investigate further